### PR TITLE
Allow users to run their own code as part of "Set Home Assistant Version" task.

### PR DESCRIPTION
**Pull request by elahd**
Adds execution hook to the "Set Home Assistant Version" script to allow users to have their own code run during the version switch process. The "set version" script will run whatever code/script is defined by the user via the POST_SET_VERSION_HOOK environment variable.

This addresses item 2 in #270 using a much more generic approach.

### CODE REVIEW

**Comment by elahd** [comment]
For context, I use this feature as follows:

1) Added the below section to my project's devcontainer.json:

```json
  "containerEnv": {
    "POST_SET_VERSION_HOOK": "/workspaces/[projectname]/post-set-version-hook.sh"
  },
```

2) Contents of `post-set-version-hook.sh`, which accepts a Home Assistant version number as an input argument:

```bash
#!/usr/bin/env bash

git_dir="/tmp/ver"
dev_dir="/ha-dev"
repo_url="https://github.com/home-assistant/core.git"

if [ -z "$1" ]; then
    echo "No version supplied. Deleting existing development files."
    rm -rf $dev_dir/pylint
    exit 1
fi

mkdir -p "$git_dir"

if [ "$(git -C "$git_dir" config --get remote.origin.url)" == "$repo_url" ]; then
    git -C "$git_dir" checkout tags/"$1"
else
    git clone "$repo_url" --branch "$version" "$git_dir"
fi

mkdir -p "$dev_dir"

if [ "$(readlink "$dev_dir"/pylint)" != "$dev_dir"/pylint ]; then
    ln -sf /tmp/ver/pylint /ha-dev/pylint
fi
```

This script checks out the Home Assistant source code for the requested HA version and makes a specific folder available within `/ha-dev`.

File: container/helpers/common/homeassistant/set-version.sh [hunk 1]
```diff
@@ -1,4 +1,11 @@
 #!/usr/bin/env bash
+# shellcheck source=/dev/null
+
+source /opt/container/helpers/common/paths.sh
```
Comment (ludeeus) [inline container/helpers/common/homeassistant/set-version.sh:4]:
```suggestion
```
Not needed

File: container/helpers/common/paths.sh [hunk 2]
```diff
@@ -6,4 +6,10 @@ function workspacePath {
     else
         echo "$(find /workspaces -mindepth 1 -maxdepth 1 -type d | tail -1)/"
     fi
+}
+
+function postSetVerionHook {
+    if [[ -n "$POST_SET_VERSION_HOOK" ]]; then
+        echo "${POST_SET_VERSION_HOOK}"
+    fi
```
Comment (ludeeus) [inline container/helpers/common/paths.sh:14]:
```suggestion
```
Not needed

File: container/helpers/common/homeassistant/set-version.sh [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
 
 read -p 'Set Home Assistant version: ' -r version
 python3 -m pip --disable-pip-version-check install --upgrade homeassistant=="$version"
+
+if [[ -n "$(postSetVerionHook)" ]]; then
+    "$(postSetVerionHook)" "$version"
```
Comment (ludeeus) [inline container/helpers/common/homeassistant/set-version.sh:10]:
```suggestion
if [[ -n "$POST_SET_VERSION_HOOK" ]]; then
    "$POST_SET_VERSION_HOOK" "$version"
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

**Comment by elahd** [comment]
@ludeeus Thanks for the feedback. I made the changes and tested them locally.

**Pull request merged**